open_humans/templates/member/my-member-research-data.html

```diff
@@ -1,89 +1,103 @@
 {% extends 'member/my-member-dashboard.html' %}
 
+{% load data_import %}
 {% load utilities %}
 
 {% block dashboard_main %}
 <div class="panel panel-default pad-all-sides">
   {% for source, task in data_retrieval_tasks.items %}
+    {% if not forloop.first %}
+      <hr class="source-divider">
+    {% endif %}
 
-  {% if not forloop.first %}
-  <hr class="source-divider">
-  {% endif %}
-  <div class="pull-right">
-    {% if task.data_files %}
-      {% if user.member.public_data_participant.enrolled %}
-      <form class="toggle-sharing"
-        action="{% url 'public-data:toggle-sharing' %}" method="POST"
-        style="display: inline-block;">
-        {% csrf_token %}
-
-        <input type="hidden" name="source" value="{{ source }}">
+    <div class="pull-right">
+      {% if task.data_files %}
+        {% source_is_connected source user as is_connected %}
 
-        {% if task.is_public %}
-          <input type="hidden" name="public" value="False">
-        {% else %}
-          <input type="hidden" name="public" value="True">
+        {% if is_connected %}
```

Comment:
oops yes, this went through a couple of iterations and the `not` got lost, good catch :)